hatrac/model/storage/amazons3.py

```diff
@@ -106,7 +124,16 @@ def create_from_file(self, name, input, nbytes, content_type=None, content_md5=N
             headers['Content-Type'] = content_type
         if content_md5 is not None:
             headers['Content-MD5'] = content_md5
-        s3_key.set_contents_from_stream(input, headers=headers, replace=True, md5=md5, size=nbytes)
+        
+        # Amazon does not support chunked transfer
+        #s3_key.set_contents_from_stream(input, headers=headers, replace=True, size=nbytes)
+        # TODO - for larger uploads we will probably need to materialize files
+        s3_key.set_contents_from_string(input.read(nbytes), headers=headers, replace=True)
+        
+        # TODO we could default to our own versioning here
```

Comment:
I think this check should be in the initial connection code to make sure the bucket has versioning turned on.  An alternate version of this entire mode (or a different configuration option?) would be to generate random version IDs and bake them into the name used at the storage layer, just like we do in the filesystem backend.  However, this would mean that hatrac names and versions do not map to S3 names and versions, which might be confusing for users who peer behind the curtain.